src/convertToObject.js

```diff
@@ -3,16 +3,32 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles
+ * (see an example in [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties
+ * (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const styleObject = {};
+  const styleArray = sourceString.split(';');
+
+  styleArray.forEach(style => {
+    const [property, value] = style.split(':');
+
+    if (property && value) {
+      const propertyName = property.trim();
+      const propertyValue = value.trim();
+
+      styleObject[propertyName] = propertyValue;
+    }
+  });
```

Comment:
suggest to replace forEach method on reduce and immediately return result of reduce
```
 return styleArray.reuduce((acc, style) => {
   // do something
 }, {})
```

Reply:
ok, i will approve but left possible solution. may be you will find what was wrong in your attemts with reduce)
it can be something like that.
```
return styleArray.reduce( (acc, style) => {
        const [key, value] = style.split(':').map(item => item.trim());

        if (!key) {
          return acc;
        }

        return {
          ...acc, [key]: value,
        };
  }, {});
```